Bottom: e1803200778d86100cf688a6f8114be88fc494c5
Top:    5f95233508d4cca7fa082a62bba862ea43f4871b
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-05-01 13:11:51 +0100

Refresh of debug-on-model-orientation

---

diff --git a/src/hid/step/step.c b/src/hid/step/step.c
index 81c11f7..a31f6c2 100644
--- a/src/hid/step/step.c
+++ b/src/hid/step/step.c
@@ -384,25 +384,25 @@ step_do_export (HID_Attr_Val * options)
         instance = g_new0 (struct assembly_model_instance, 1);
         instance->name = NAMEONPCB_NAME (element);
 #ifdef REVERSED_PCB_CONTOURS
-        instance->ox = COORD_TO_STEP_X (PCB, element->MarkX) +                    ( ox * cos_rot + oy * sin_rot),
-        instance->oy = COORD_TO_STEP_Y (PCB, element->MarkY) - on_solder_negate * (-ox * sin_rot + oy * cos_rot),
-        instance->oz = (on_solder ? -COORD_TO_STEP_Z (PCB, HACK_BOARD_THICKNESS) : 0.0) - on_solder_negate * oz;
-        instance->ax =                    ( ax * cos_rot + ay * sin_rot),
-        instance->ay = -on_solder_negate * (-ax * sin_rot + ay * cos_rot),
-        instance->az = on_solder_negate * az;
-        instance->rx =                    ( rx * cos_rot + ry * sin_rot), /* XXX: SHOULD THIS FACTOR IN on_solder_negate? */
-        instance->ry = -on_solder_negate * (-rx * sin_rot + ry * cos_rot), /* XXX: SHOULD THIS FACTOR IN ol_solder_negaet? */
-        instance->rz = on_solder_negate * rz;
+        instance->ox =                     ( ox * cos_rot + oy * sin_rot) + COORD_TO_STEP_X (PCB, element->MarkX);
+        instance->oy = -on_solder_negate * (-ox * sin_rot + oy * cos_rot) + COORD_TO_STEP_Y (PCB, element->MarkY);
+        instance->oz =  on_solder_negate * oz                             + (on_solder ? -COORD_TO_STEP_Z (PCB, HACK_BOARD_THICKNESS) : 0.0);
+        instance->ax =                     ( ax * cos_rot + ay * sin_rot);
+        instance->ay = -on_solder_negate * (-ax * sin_rot + ay * cos_rot);
+        instance->az =  on_solder_negate * az;
+        instance->rx =                     ( rx * cos_rot + ry * sin_rot); /* XXX: SHOULD THIS FACTOR IN on_solder_negate? */
+        instance->ry = -on_solder_negate * (-rx * sin_rot + ry * cos_rot); /* XXX: SHOULD THIS FACTOR IN ol_solder_negaet? */
+        instance->rz =  on_solder_negate * rz;
 #else
-        instance->ox = COORD_TO_STEP_X (PCB, element->MarkX) +                    ( ox * cos_rot + oy * sin_rot),
-        instance->oy = COORD_TO_STEP_Y (PCB, element->MarkY) + on_solder_negate * (-ox * sin_rot + oy * cos_rot),
-        instance->oz = on_solder_negate * -COORD_TO_STEP_Z (PCB, HACK_BOARD_THICKNESS) / 2.0 + on_solder_negate * oz;
-        instance->ax =                    ( ax * cos_rot + ay * sin_rot),
-        instance->ay = on_solder_negate * (-ax * sin_rot + ay * cos_rot),
-        instance->az = on_solder_negate * az;
-        instance->rx =                    ( rx * cos_rot + ry * sin_rot), /* XXX: SHOULD THIS FACTOR IN on_solder_negate? */
-        instance->ry = on_solder_negate * (-rx * sin_rot + ry * cos_rot), /* XXX: SHOULD THIS FACTOR IN ol_solder_negaet? */
-        instance->rz = on_solder_negate * rz;
+        instance->ox =                     ( ox * cos_rot + oy * sin_rot) + COORD_TO_STEP_X (PCB, element->MarkX);
+        instance->oy =  on_solder_negate * (-ox * sin_rot + oy * cos_rot) + COORD_TO_STEP_Y (PCB, element->MarkY);
+        instance->oz =  on_solder_negate * oz                             + on_solder_negate * -COORD_TO_STEP_Z (PCB, HACK_BOARD_THICKNESS) / 2.0;
+        instance->ax =                     ( ax * cos_rot + ay * sin_rot);
+        instance->ay =  on_solder_negate * (-ax * sin_rot + ay * cos_rot);
+        instance->az =  on_solder_negate * az;
+        instance->rx =                     ( rx * cos_rot + ry * sin_rot); /* XXX: SHOULD THIS FACTOR IN on_solder_negate? */
+        instance->ry =  on_solder_negate * (-rx * sin_rot + ry * cos_rot); /* XXX: SHOULD THIS FACTOR IN ol_solder_negaet? */
+        instance->rz =  on_solder_negate * rz;
 #endif
         model->instances = g_list_append (model->instances, instance);
